tests/test_app/test_data_store.py

```diff
@@ -276,6 +278,36 @@ class DirectoryDataStoreTests(TestCase, DataStoreBaseTests):
     ReadClass = ReadOnlyDirectoryDataStore
     WriteClass = WritableDirectoryDataStore
 
+    def test_write_class_source_create_delete(self):
+        with TemporaryDirectory(dir=".") as dirname:
+            # tests the case when the directory has the file with the same suffix to self.suffix
+            path = os.path.join(dirname, "delme_dir")
+            os.mkdir(path)
+            with open(
+                os.path.join(path, "test_write_class_source_create_delete.json"), "w"
+            ):
+                pass
+            dstore = self.WriteClass(
+                path, suffix=".json", if_exists=OVERWRITE, create=True
+            )
+            self.assertEqual(len(glob.glob(os.path.join(path, "*.json"))), 0)
```

Comment:
pretty sure this can be replaced with `self.assertEqual(len(dstore), 0)`, so you can remove the glob import and calls.